frame/staking/src/pallet/mod.rs

```diff
@@ -880,7 +886,19 @@ pub mod pallet {
 
 				// Note: in case there is no current era it is fine to bond one era more.
 				let era = Self::current_era().unwrap_or(0) + T::BondingDuration::get();
-				ledger.unlocking.push(UnlockChunk { value, era });
+				if let Some(mut chunk) =
+					ledger.unlocking.last_mut().filter(|chunk| chunk.era == era)
+				{
+					// To keep the chunk count down, we only keep one chunk per era. Since
+					// `unlocking` is a FiFo queue, if a chunk exists for `era` we know that it will
+					// be the last one.
+					chunk.value = chunk.value.saturating_add(value)
+				} else {
+					ledger
+						.unlocking
+						.try_push(UnlockChunk { value, era })
+						.map_err(|_| Error::<T>::NoMoreChunks)?;
```

Comment:
This can only happen when the configured `MaxUnlockingChunks` is wrong?

Reply:
By time we get here this should never happen because of the check at the top of this function:

```rust
			ensure!(
				ledger.unlocking.len() < MaxUnlockingChunks::get() as usize,
				Error::<T>::NoMoreChunks,
			);
```

If we removed that check, this could still error in the situation where the user already has `MaxUnlockingChunks` - which would actually be fine since we haven't written to storage yet by time we get here.